doc/operator-capabilities.md

```diff
@@ -0,0 +1,193 @@
+# Operator Capability Levels
+
+Operators come in different maturity levels in regards to their lifecycle management capabilities for the application or workload they deliver. The capability models aims to provide guidance in terminology to express what features users can expect from an operator.
```

Comment:
These lifecycle management capabilities are related to the integration with OLM and the operator lifecycle on it I think we should make it clear.

Reply:
I disagree. Abilities to lifecycle the Operand across versions and perform advanced Operations on it as well as using Operand metrics to drive are in the domain of the Operator itself. They are not strongly related to the integration with OLM. They are there to inspire the author to think about the feature set of their Operator. The SDK does provide features to achieve these higher levels, e.g. scaffold metrics endpoints and (soon) webhooks.
I agree the SDK would be the place where to put more tools and helpers to achieve higher capability levels.